### PR TITLE
docs: add "why did my mutant survive?" troubleshooting section

### LINKED ISSUE #416: Docs note: mutating functions referenced by Pydantic default_factory (or any closure-captured callable) is invisible to the model
Spent a while debugging surviving mutants and traced it to a Python-isn't-magic detail worth surfacing in docs.

```python
def _now() -> datetime:
    return datetime.now(tz=UTC)

class HistoryEntry(BaseModel):
    timestamp: datetime = Field(default_factory=_now)
```

Gremlins mutates `_now`'s return to `None`. The mutation patches the module-level `_now` name, but Pydantic captures the function reference at class-define time inside `Field(default_factory=_now)`. When `HistoryEntry()` is instantiated, the original `_now` still runs — the mutant is invisible to the model.

Same pattern affects any callable captured at import time:

Pydantic `default_factory=fndataclasses.field(default_factory=fn)` Decorator registration at import (e.g., `@app.route(...))functools.partial(fn, ...)` stored in module-level constants

Not a bug — it's how Python references work — but it's a non-obvious survival pattern that trips up anyone running gremlins on Pydantic-heavy or registry-heavy code.

Proposal: a "Why did my mutant survive?" docs section listing these patterns and the workaround: test the mutated function directly (assert _now() is not None) instead of testing it only through the model that captured it. Could be a single page linked from README under Troubleshooting.

### PR DESCRIPTION
**Pull request by mikelane**
## Summary

Adds a **"Why did my mutant survive?"** section to `docs/guide/troubleshooting.md`, documenting a non-obvious Python gotcha reported in #416: mutating a function referenced by a **closure-captured callable** is invisible to pytest-gremlins' mutation model.

## The gotcha

```python
def _now() -> datetime:
    return datetime.now(tz=UTC)

class HistoryEntry(BaseModel):
    timestamp: datetime = Field(default_factory=_now)
```

Gremlins mutates `_now`'s return to `None` by rebinding the module-level `_now` **name**. But `Field(default_factory=_now)` captured the function **object** at class-definition (import) time — so the mutation and the captured reference never meet, and the survivor is invisible. The section explains the mechanism concretely and generalizes to the other import-time capture patterns (`dataclasses.field(default_factory=...)`, import-time decorator registration, module-level `functools.partial`).

**Workaround documented:** cover small factory/helper functions with their own direct tests (`assert _now() is not None`) in addition to integration tests.

## Changes

- `docs/guide/troubleshooting.md` — new section (placed before "Getting Help")
- `cspell.json` — added `pydantic` to the dictionary

Docs-only, no executable code. `mkdocs build --strict` passes.

Closes #416

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "pydocstyle",
     "pyenv",
     "pylint",
+    "pydantic",
```
Comment (mikelane) [inline cspell.json:126]: Nit (LOW, non-blocking): `pydantic` is inserted after `pylint`/before `pymdownx`, but alphabetically it belongs between `pycache` and `pydocstyle` (comparing `pyda` vs `pydo`, 'a' < 'o'). The dictionary wasn't strictly alphabetized before this PR either, so this isn't a new regression — just a small readability nit for a future cleanup pass. Doesn't affect cspell's behavior.

**Review by mikelane — COMMENTED** [review]
## Review: docs-only, "Why did my mutant survive?" troubleshooting section

**Scope confirmed docs-only**: `docs/guide/troubleshooting.md` (new section) + `cspell.json` (added `pydantic`). No executable code changed. Adversarial QA gauntlet artifact correctly marked `"gate": "exempt"`, `"status": "pass"`, `head_sha` matches.

### Technical accuracy — verified by reproduction, not just reading

I reproduced both halves of the doc's claim against the actual plugin (not the docs prose) using a real pytester run with `--gremlins --gremlin-targets=src_module.py` against the exact `HistoryEntry`/`Field(default_factory=_now)` example from the PR:

- **Without a direct test of `_now`** (only a test that instantiates `HistoryEntry` and asserts `timestamp is not None`): the gremlin on `_now`'s return statement **survives** (`Survived: 1 gremlins`).
- **With a direct test added** (`assert _now() is not None`): the same gremlin **dies** (`Survived: 0 gremlins`).

I also inspected the raw instrumented AST (`ast.unparse` on `transform_source(..., [ReturnOperator()])`) to confirm the mechanism: pytest-gremlins rewrites `_now`'s body to switch on the module-global `__gremlin_active__`, and `Field(default_factory=_now)` captures the function object at class-definition time. The doc's simplification ("name vs. object") is a reasonable and accurate way to explain this to users — the observable behavior (survives without direct test, dies with one) matches exactly what's described.

The generalization list (`dataclasses.field(default_factory=...)`, import-time decorator registration, `functools.partial` at module scope) is a correct extension of the same "reference captured once at import time" principle — no code-level verification needed there since it follows directly from CPython's late-binding-of-names-vs-early-binding-of-references semantics, which the `_now` case already demonstrates concretely.

### Prose / voice

Matches the existing troubleshooting.md structure (Symptom / Cause / Solution, `---` section dividers, admonition-free plain prose). Domain terms (survivor, gremlin, zap) used correctly and consistently with the rest of the doc. The code example is valid, runnable Python (verified — it imports and instantiates cleanly under Python 3.11+/Pydantic 2.x).

### Build verification

`uv run mkdocs build --strict` — exit 0, no warnings promoted to errors (existing "not in nav" INFO messages are pre-existing and unrelated to this PR's file).

### Regression check (squash-merge hazard)

Divergence point checked (`git merge-base`). Files touched on `main` since divergence (`src/pytest_gremlins/plugin.py`, `tests/plugin/test_plugin_helpers.py`) do **not** overlap with this PR's changed files. No regression risk.

### Findings

- **LOW** (posted inline, `cspell.json:126`): `pydantic` inserted after `pylint` instead of its alphabetical position between `pycache` and `pydocstyle`. Pre-existing list wasn't strictly sorted before this PR either, so not a new regression — just a nit for a future cleanup pass.

No CRITICAL, HIGH, or MEDIUM findings.

### Verdict

**APPROVE.** Verdict artifact written to `~/second-brain/06_Agent/artifacts/pr-review/mikelane__pytest-gremlins__457.json` with `scope: "docs-only"`. Posting as `COMMENT` rather than `APPROVE` event since this is an owner-authored PR (GitHub disallows self-approval) — the artifact is the authoritative gate for merge-guardian.
